### PR TITLE
Removes the ability for inducers and circuits to charge energy weapons.

### LINKED ISSUE #37109: Inducer can be used to recharge weapons with integrated power cells
Which means pretty much every single energy gun. Allegedly this includes advanced energy guns (can only self-charge) and, by extension, plasma cutters (plasma recharge only).

### PR DESCRIPTION
**Pull request by Jalleo**
Fixes a power creep that occurred with two objects that weren't known to @KorPhaeron at the time they were merged and now needs reverting.

Fixes #37109 

:cl: Spoilsport
del: Inducers and integrated circuits can no longer charge energised weapons.
/:cl:

### CODE REVIEW

File: code/game/objects/items/inducer.dm [hunk 1]
```diff
@@ -102,12 +102,11 @@
 	else
 		recharging = TRUE
 	var/obj/item/stock_parts/cell/C = A.get_cell()
-	var/obj/item/gun/energy/E
 	var/obj/O
 	var/coefficient = 1
 	if(istype(A, /obj/item/gun/energy))
-		coefficient = 0.075 // 14 loops to recharge an egun from 0-1000
-		E = A
+		to_chat(user,"Error unable to interface with device"
```
Comment (Jalleo) [inline code/game/objects/items/inducer.dm]: Fuck I forgot to bracket it

**Comment by DaxDupont** [comment]
is this an exploit?

**Review by Cyberboss — APPROVED** [review]

**Comment by KorPhaeron** [comment]
Having to return to a safe area and charge your guns has been the drawback of energy weapons for the entire time our server has existed and I highly doubt anyone is gonna convince me a roundstart item granting your guns unlimited ammo in the field is a good idea in the next 24 hours

**Comment by evsey9** [comment]
is there any way whatsoever for integrated circuits being able to charge their gun that this PR doesn't remove?
because having to disassemble your circuit, remove the weapon circuit, put the gun in a charger, then put the gun back into the component, and then put the component back into the circuit and then have to rewirte the component seems VERY tedious and annoying.

**Comment by Jalleo** [comment]
I will check if there is tomorrow.

**Comment by evsey9** [comment]
Coding in an ability for circuits to only recharge held guns would be nice. Or, perhaps, ability to intterface with rechargers.

**Comment by PKPenguin321** [comment]
Yeah, would like this to be kept for ICs at least until they work with cell chargers

**Comment by ma44** [comment]
To add to that, a fully upgraded recharger should be able to recharge any advanced taser/energy guns in only one process tick.

**Comment by arsserpentarium** [comment]
This change renders weapon firing circuits completely unusable.
Also it isn't exploit.Inducers was able to charge guns for very long time.Kor can't not know it.

**Comment by Poojawa** [comment]
Inducers were coded specifically to make weapon rechargers obsolete under the guise of "charging APCs"

so them not being compatible at all to recharge your backpack armory with bluespace/slime cells is a good thing.

**Comment by Cyberboss** [comment]
We don't negotiate with terrorists. This will make someone angry enough to add recharger compatibility to circuits

**Comment by evsey9** [comment]
Ok then this will make me angry enough

**Pull request merged**